src/python/netvlad/netvlad_extractor.py

```diff
@@ -0,0 +1,217 @@
+# This file is highly inspired by
+#  https://github.com/cvg/Hierarchical-Localization/blob/master/hloc/extractors/netvlad.py
+# The weights are downloaded from https://cvg-data.inf.ethz.ch/hloc/netvlad/Pitts30K_struct.mat
+# The parsing of the .mat file is according to aforementioned project.
+
+import torch
+from torch import nn
+from torchvision.transforms import ToTensor
+import torchvision.models as models
+from torch.utils.data import Dataset
+
+from PIL import Image
+
+from scipy.io import loadmat
+import numpy as np
+
+from pathlib import Path
+import argparse
+
+
+class ImageDataset(Dataset):
+    def __init__(self, img_dir, transform=None):
+        self.img_names = []
+        self.img_names.extend(img_dir.glob("*.png"))
+        self.img_names.extend(img_dir.glob("*.jpg"))
+        self.img_names.sort()
+        self.transform = transform
+
+    def __len__(self):
+        return len(self.img_names)
+
+    def __getitem__(self, idx):
+        assert idx >= 0 and idx < len(self.img_names)
+        image = Image.open(self.img_names[idx].as_posix())
+        image = image.resize((224, 224))
```

Comment:
Magic constant detected! I'd really like to see this number as a named constant somewhere so that it provides a little more insight into what it represents and why it is like this.

Reply:
Added a constant.